inventories/README.md

```diff
@@ -0,0 +1,309 @@
+Setting Up Gateway Inventory
+=======
+Currently gateway inventory file is using `yaml` style format. File extension can be either `yaml` or `yml`. It follows the *alternative folder structure* defined in [Ansible Best Practice](https://docs.ansible.com/ansible/latest/user_guide/playbooks_best_practices.html#alternative-directory-layout)
+
+## Host Machine Mapping
+To use the same inventory for upgrading gateway, each target host machine will need to have `src` defined, like below.
+
+```yml
+gateway_primary_db:
+  hosts:
+    target-gw.placeholder.com:
+      src: source-db-gw.placeholder.com
+```
+Without src defined, playbooks and role used for gateway upgrading will not function. ex. [Gateway Preupgrade Analyzer](../roles/gateway_preupgrade_analyzer)
+
+By following this convention, new playbook for target machine can be used against the same inventory file without modification. Save time and effort on code maintainance.
+
+## Host Groups
+> gateway //master group contains all gateway subgroups
+
+> gateway_mysql //master group contains all gateway mysql subgroups
+
+> gateway_primary_db //group contains gateway master database node
+
+> gateway_failover_db //group contains gateway failover database node
+
+> gateway_processing_node //group contains gateway node process traffic only, no database running.
+
+## Current Sample Inventory 
+
+To view the current sample inventory in a tree view, `ansible-inventory` command can be used.
+
+```bash
+ansible-inventory --graph -i ./inventory/sample/hosts.yml
+```
+Output:
+```
+@all:
+  |--@gateway:
+  |  |--@gateway_hsm:
+  |  |--@gateway_mysql:
+  |  |  |--@gateway_failover_db:
+  |  |  |  |--failover-gw.placeholder.com
+  |  |  |--@gateway_primary_db:
+  |  |  |  |--target-gw.placeholder.com
+  |  |--@gateway_papim_install:
+  |  |--@gateway_papim_source:
+  |  |--@gateway_processing_node:
+  |  |  |--target-processing-gw.placeholder.com
+  |--@hsm:
+  |  |--@hsm_luna7:
+  |  |  |--luna7.placeholder.com
+  |--@ungrouped:
+```
+
+For more detailed view, use option `--list`
+```bash
+ansible-inventory --list -i ./inventory/sample/hosts.yml
+```
+Output:
+```
+{
+    "_meta": {
```

Comment:
Is it necessary to provide the out put? or maybe collapse it.

Reply:
collapse it is a good idea. i will try to add that

Reply:
updated with collapsible section